pkg/devfile/parser/parse.go

```diff
@@ -49,6 +50,11 @@ func parseDevfile(d DevfileObj, resolveCtx *resolutionContextTree, tool resolver
 		return d, errors.Wrapf(err, "failed to decode devfile content")
 	}
 
+	err = parseKubeComponentFromURI(d)
```

Comment:
does it make sense to parse after flattening in case there are parent overrides for the kubernetes uri?

Reply:
my bad, uri should not be converted to inlined until flattenned. updated